### PR TITLE
Ecodiv techno

### PR DESCRIPTION
**Pull request by LiuDilongNJ**

### CODE REVIEW

File: init.sql [hunk 1]
```diff
@@ -118,6 +122,7 @@ CREATE TABLE `file_upload`
     `rating`         enum('A','B','C','D','E') COLLATE utf8_unicode_ci DEFAULT NULL,
     `type`           varchar(50) COLLATE utf8_unicode_ci           DEFAULT NULL,
     `medium`         varchar(50) COLLATE utf8_unicode_ci           DEFAULT NULL,
+    `recording_gain`        int(11) DEFAULT NULL,
```
Comment (shofstet) [inline init.sql:125]: The recording gain has to be added when I upload the sound file!

File: src/src/BioSounds/Controller/Administration/RecordingController.php [hunk 2]
```diff
@@ -104,6 +109,13 @@ public function save()
             throw new ForbiddenException();
         }
 
+        if (!preg_match('/^[0-9]\d*$/', $_POST['recording_gain_number'])) {
+            return json_encode([
+                'isValid' => 1,
+                'message' => 'Recording gain must be a positive integer.',
+            ]);
+        }
+
```
Comment (shofstet) [inline src/src/BioSounds/Controller/Administration/RecordingController.php:118]: Here the code where the recording gain has to be set as positiv integer!

**Review by shofstet — APPROVED** [review]
I've installed the new branch and it works well. 
I'm actually not sure how you manage the db changes on the production system. Are there any migration techniques?

File: src/src/BioSounds/Controller/Administration/RecordingController.php (hunk 2 above)
Comment (kdarras) [reply src/src/BioSounds/Controller/Administration/RecordingController.php:118]: OK I corrected this, thanks

File: init.sql (hunk 1 above)
Comment (kdarras) [reply init.sql:125]: so, the default here is NULL but on production we require a gain value... does this file's code need to be updated @LiuDilongNJ ?

**Comment by LiuDilongNJ** [comment]
I put a restriction on the recording_gain field on the server side so that it doesn't appear to be null. Since we didn't have a recording_gain before, and the recording_gain of the previous recording wasn't always 0, I kept the null case.
The production database is deployed using the structural synchronisation feature.

**Pull request merged**